plugins/modules/sda_fabric_devices_workflow_manager.py

```diff
@@ -4022,6 +4022,55 @@ def get_diff_merged(self, config):
 
         return self
 
+    def arrange_config_for_deletion(self, device_config):
+        """
+        Arrange the device config in such a way that the device with the role 'CONTROL_PLANE_NODE'
+        will be deleted at last.
+
+        Parameters:
+            device_config (list): Devices config details provided by the user in the playbook.
+        Returns:
+            updated_device_config (list): Updated device config details.
+        Description:
+            Loop in the device config information. If the device doesnot exists, prepend the
+            config details and the information collected from the Cisco Catalyst Center to the
+            new list or if the device has a role of 'CONTROL_PLANE_NODE', append the config details
+            and the information collected from the Cisco Catalyst Center to the new list.
+            Update the self.have.get("fabric_details") and return the new device_config list.
+        """
+
+        fabric_device_index = -1
+        updated_device_config = []
+        update_have = []
```

Comment:
self.log("Starting to reorder devices based on their existence and role.")
    self.log("Input device_config: {}".format(device_config))